### ISSUE TITLE
nixos-rebuild test removing IP addresses from bridge

### ISSUE THREAD

**Issue by Ekleog** [NixOS/nixpkgs#23241]
## Issue description

A bridge with configured IP address loses it when running `nixos-rebuild switch` with another bridge config. I guess it is some dependency error, as after a reboot the bridge gets back its IP address. Not using networkd.

### Steps to reproduce

Add to your configuration.nix
```
    networking.bridges.test = { interfaces = [ ]; };
    networking.interfaces.test = {
     ip4 = [ { address = "192.168.42.42"; prefixLength = 24; } ];
    };
```
Run `nixos-rebuild switch`, reboot. The bridge "test" gets an IP address assigned.

Add anything to the `interfaces` parameter of the bridge config.  Run `nixos-rebuild switch`. The bridge no longer has an IP.

## Technical details

* System: nixos-unstable (branched off a week ago from 183eeb3c0fdac8de3146aedaa6028b474f96db6f for work on a module, can test from upstream if need be)
* Nix version: 1.11.6
* Nixpkgs version: 17.03pre101839.53a2baa

**Comment by Ekleog** [comment]
I have just been pointed to https://github.com/NixOS/nixpkgs/issues/16230 ; it's weird because it's not the same issue (the bridge *is* created, for my case), but a similar one. Maybe it has been hot-patched as advised in the comments, yet the underlying issue is the same?

**Comment by globin** [comment]
Probably duplicate of #21745 and possible fix in #22850

**Comment by globin** [comment]
Should be fixed.
